man/fi_lnx.7.md

```diff
@@ -0,0 +1,156 @@
+  structure to select the interface to use. A common selection criteria is
+  the interface nearest the core the process is bound to. In order to make
+  this determination, the application requires the PCI information about the
+  interface. For this reason LNX forwards the PCI information for the
+  inter-node provider in the link to the application.
+
+# LIMITATIONS AND FUTURE WORK
+
+*Hardware Support*
+: LNX doesn't support hardware offload; ex hardware tag matching. This is
+  an inherit limitation when using the peer infrastructure. Due to the use
+  of a shared receive queue which linked providers need to query when
+  a message is received, any hardware offload which requires sending the
+  receive buffers to the hardware directly will not work with the shared
+  receive queue. The shared receive queue provides two advantages; 1) reduce
+  memory usage, 2) coordinate the receive operations. For #2 this is needed
+  when receiving from FI_ADDR_UNSPEC. In this case both providers which are
+  part of the link can race to gain access to the receive buffer. It is
+  a future effort to determine a way to use hardware tag matching and other
+  hardware offload capability with LNX
+
+*Limited Linking*
+: This release of the provider supports linking SHM provider for intra-node
+  operations and another provider which supports the FI_PEER capability for
+  inter-node operations. It is a future effort to expand to link any
+  multiple sets of providers.
+
+*Memory Registration*
+: As part of the memory registration operation, varying hardware can perform
+  hardware specific steps such as memory pinning. Due to the fact that
+  memory registration APIs do not specify the source or destination
+  addresses it is not possible for LNX to determine which provider to
+  forward the memory registration to. LNX, therefore, registers the memory
+  with all linked providers. This might not be efficient and might have
+  unforeseen side effects. A better method is needed to support memory
+  registration.
+
+*Operation Types*
+: This release of LNX supports tagged and RMA operations only. Future
+  releases will expand the support to other operation types.
+
+*Multi-Rail*
+: Future design effort is being planned to support utilizing multiple interfaces
+  for traffic simultaneously. This can be over homogeneous interfaces or over
+  heterogeneous interfaces.
+
+# RUNTIME PARAMETERS
+
+The *LNX* provider checks for the following environment variables:
+
+*FI_LNX_PROV_LINKS*
+: This environment variable is used to specify which providers to link. This
+  must be set in order for the LNX provider to return a list of fi_info
+  blocks in the fi_getinfo() call. The format which must be used is:
+  <prov1>+<prov2>+... As mentioned earlier currently LNX supports linking
+  only two providers the first of which is SHM followed by one other
+  provider for inter-node operations
+
+*FI_LNX_DISABLE_SHM*
```

Comment:
How is `FI_LNX_PROV_LINKS=cxi` different from `FI_LNX_PROV_LINKS=shm+cxi` with `FI_LNX_DISABLE_SHM=1`?

Reply:
At the moment you can't do FI_LNX_PROV_LINKS=cxi. It has to be of the format shm+<inter node provider>. I would argue as well that it doesn't make sense to allow setting that env var to just a single provider. By definition it identifies a link and you can't have a link composed of one provider. Now we might need to revisit that when we consider multi-rail since you can link multiple interfaces. But I would rather postpone this to a later point.

Reply:
I'm ok revisiting later.